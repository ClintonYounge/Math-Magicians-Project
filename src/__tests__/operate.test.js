import operate from '../logic/operate';

describe('test operate', () => {
  it('subtract numbers', () => {
    const test = operate(10, 4, '-');
    expect(test).toMatchSnapshot('6');
  });

  it('plus two numbers', () => {
    const test = operate(10, 8, '+');
    expect(test).toMatchSnapshot('18');
  });

  it('times two numbers', () => {
    const test = operate(3, 7, 'x');
    expect(test).toMatchSnapshot('21');
  });

  it('divide two numbers', () => {
    const test = operate(15, 3, 'รท');
    expect(test).toMatchSnapshot('5');
  });

  it('divide a number by 0', () => {
    const test = operate(19, 0, 'รท');
    expect(test).toMatchSnapshot("Can't divide by 0.");
  });

  it('Module a number by 0', () => {
    const test = operate(19, 0, '%');
    expect(test).toMatchSnapshot("Can't find modulo as can't divide by 0.");
  });

  it('Module a number by 0', () => {
    const test = operate(8, 3, '%');
    expect(test).toMatchSnapshot('2');
  });
});
