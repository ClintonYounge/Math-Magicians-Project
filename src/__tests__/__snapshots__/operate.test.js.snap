// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test operate Module a number by 0: 2 1`] = `"2"`;

exports[`test operate Module a number by 0: Can't find modulo as can't divide by 0. 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`test operate divide a number by 0: Can't divide by 0. 1`] = `"Can't divide by 0."`;

exports[`test operate divide two numbers: 5 1`] = `"5"`;

exports[`test operate plus two numbers: 18 1`] = `"18"`;

exports[`test operate subtract numbers: 6 1`] = `"6"`;

exports[`test operate times two numbers: 21 1`] = `"21"`;
